Library/Homebrew/utils/shell_env.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+set +o posix
+
+KeepKeys=(
+  BASH
+  BASH_ARGC
+  BASH_ARGV
+  BREW_FILE_DIRECTORY
+  USR_LOCAL_BREW_FILE_DIRECTORY
+  USR_LOCAL_HOMEBREW_REPOSITORY
+  HOMEBREW_*
+  HOME
+  PWD
+  LANG
+  PATH
+  TERM
+)
+
+prune-shell-env() {
+  file=/tmp/homebrew-env
```

Comment:
Actually was thinking about this more and I think the ideal location for this would be just in a `HOMEBREW_OLD_ENV` variable that can be read later so e.g. `export HOMEBREW_OLD_ENV="$(env)"`. Thoughts?